modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/scheduler/jobs/ReplicationFactorCheckJob.java

```diff
@@ -0,0 +1,230 @@
+            // if a storage node does not have a resource, then that means it was was
+            // deployed prior to installing the server.
+            if (storageNode.getOperationMode() == StorageNode.OperationMode.NORMAL
+                || storageNode.getOperationMode() == StorageNode.OperationMode.MAINTENANCE
+                || storageNode.getResource() == null) {
+                hostNames.add(storageNode.getAddress());
+            }
+        }
+        if (hostNames.isEmpty()) {
+            log.error("There is not storage node in relational database to connect! Please re-install at least 1 storage node");
+            return;
+        }
+        debug("Initiating connection to cluster hosts=" + Arrays.toString(hostNames.toArray()));
+        int port = storageNodes.get(0).getCqlPort();
+
+        Cluster cluster = new ClusterBuilder().addContactPoints(hostNames.toArray(new String[hostNames.size()]))
+            .withCredentialsObfuscated(username, password).withPort(port)
+            .withLoadBalancingPolicy(new RoundRobinPolicy())
+            .withRetryPolicy(new LoggingRetryPolicy(DefaultRetryPolicy.INSTANCE))
+            .withCompression(ProtocolOptions.Compression.NONE).build();
+
+        // need to connect to system keyspace as it contains data about other keyspaces (and replication factor)
+        Session session = null;
+        try {
+            session = cluster.connect("system");
+            debug("Querying system keyspaces for strategy_options");
+            Map<String, Integer> replicationFactors = getReplicationFactors(session);
+
+            if (replicationFactors.size() != KEYSPACES.size()) {
+                log.error("Failed to query storage cluster for keyspaces " + Arrays.toString(KEYSPACES.toArray())
+                    + " for replication_factor, expected to retrieve " + KEYSPACES.size() + " but got "
+                    + replicationFactors.size());
+
+                return;
+            }
+            Map<String, Integer> factorsToSet = new HashMap<String, Integer>(replicationFactors.size());
+            String inconsistency = "";
+            for (Entry<String,Integer> factorEntry : replicationFactors.entrySet()) {
+                int healthy = getHealthyReplicationFactor(factorEntry.getKey(), storageNodes.size());
+                int current = factorEntry.getValue().intValue();
+
+                if (current != healthy) {
+                    inconsistency += "keyspace [" + factorEntry.getKey() + "] has replication_factor=" + current
+                        + " but it should be " + healthy + ", ";
+                    factorsToSet.put(factorEntry.getKey(), healthy);
+                }
+            }
+
+            if (!factorsToSet.isEmpty()) {
+                // strip last ", "
+                inconsistency = inconsistency.substring(0, inconsistency.length() - 2);
+                log.error("Storage Cluster is not consistent! There are "
```

Comment:
I might reduce the log level to WARN since this is a situation which we can handle or at least try to handle. In the log statement in addition to the number of storage nodes in the RDBMS, it would be helpful to also include the current RF and the expected RF.

Lastly, there is misspelling in rational.